### PR TITLE
feat: Asana blocker fixes

### PR DESCRIPTION
**Pull request by akshbhu**
#### Description of changes
- Added userPool Group Override.ts file
-  AuthRole//UnAuthRole/Deplyment bucket changes to team ProviderInfo file
- Remmove node_modules during push from custom category

#### Description of how you validated changes
- Tested manually

#### Checklist

- [X] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-auth/resources/overrides-resource/userPoolGroups/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "name": "overrides-for-UserPoolGroup-resource",
```
Comment (kaustavghosh06) [inline packages/amplify-category-auth/resources/overrides-resource/userPoolGroups/package.json:2]: change to just "overrides"

File: packages/amplify-category-auth/resources/schemas/awsCognito/CognitoCLIInputs.schema.json [hunk 2]
```diff
@@ -93,7 +93,7 @@
           }
         },
         "userpoolClientRefreshTokenValidity": {
-          "type": "number"
+          "type": ["string", "number"]
```
Comment (kaustavghosh06) [inline packages/amplify-category-auth/resources/schemas/awsCognito/CognitoCLIInputs.schema.json:96]: Should it just be "number" or does it accept both string and number?

File: packages/amplify-category-auth/src/commands/auth/override.ts [hunk 3]
```diff
@@ -18,14 +18,28 @@ export const run = async (context: $TSContext) => {
...
+    selectedAuthResource = authResources.filter(resource => resource !== 'userPoolGroups')[0];
+    userPoolGroupResource = 'userPoolGroups';
+  }
   await checkAuthResourceMigration(context, selectedAuthResource);
-  const backendDir = pathManager.getBackendDirPath();
 
-  const destPath = path.normalize(path.join(backendDir, category, selectedAuthResource));
-  const srcPath = path.normalize(path.join(__dirname, '..', '..', '..', 'resources', 'overrides-resource'));
+  // override structure for auth resource
+  await generateOverrideforAuthResource(context, selectedAuthResource, 'auth');
```
Comment (kaustavghosh06) [inline packages/amplify-category-auth/src/commands/auth/override.ts:31]: wouldn't this be required to be wrapped around an if as well?

File: packages/amplify-cli/src/extensions/amplify-helpers/update-amplify-meta.ts [hunk 4]
```diff
@@ -64,7 +64,7 @@ function moveBackendResourcesToCurrentCloudBackend(resources: $TSObject[]) {
     // in the case that the resource is being deleted, the sourceDir won't exist
     if (fs.pathExistsSync(sourceDir)) {
       fs.copySync(sourceDir, targetDir);
-      if (resource?.service === ServiceName.LambdaFunction) {
+      if (resource?.service === ServiceName.LambdaFunction || resource?.service.includes('custom')) {
```
Comment (kaustavghosh06) [inline packages/amplify-cli/src/extensions/amplify-helpers/update-amplify-meta.ts:67]: should this be category or service?

Comment (kaustavghosh06) [reply packages/amplify-cli/src/extensions/amplify-helpers/update-amplify-meta.ts:67]: Please make sure push/pull works with this. Also is there hashing logic for node_modules when running amplify-status?

File: packages/amplify-category-auth/resources/schemas/awsCognito/CognitoCLIInputs.schema.json (hunk 2 above)
Comment (akshbhu) [reply packages/amplify-category-auth/resources/schemas/awsCognito/CognitoCLIInputs.schema.json:96]: it accepts both , in CFN its passed as number

File: packages/amplify-cli/src/extensions/amplify-helpers/update-amplify-meta.ts (hunk 4 above)
Comment (akshbhu) [reply packages/amplify-cli/src/extensions/amplify-helpers/update-amplify-meta.ts:67]:
``` 
 "custom": {
    "customResource9bcf0700": {
      "service": "customCDK",
      "providerPlugin": "awscloudformation",
      "providerMetadata": {
        "s3TemplateURL": "https://s3.amazonaws.com/amplify-nodemodulesremove-dev-195034-deployment/amplify-cfn-templates/custom/customResource9bcf0700-cloudformation-template.json",
        "logicalId": "customcustomResource9bcf0700"
      },
```

File: packages/amplify-category-auth/src/commands/auth/override.ts (hunk 3 above)
Comment (akshbhu) [reply packages/amplify-category-auth/src/commands/auth/override.ts:31]: no , if no resource is selected it return and 
if authResource selected -> skeleton generated for auth resource

if userPoolGroupResource selected -> skeleton generated for auth resource and userPool Group resource

Comment (kaustavghosh06) [reply packages/amplify-category-auth/src/commands/auth/override.ts:31]: when userPoolGroupResource is selected why do we need to generate override for both auth resource and userpool group resource and not just user pool group resource?

**Review by kaustavghosh06 — APPROVED** [review]

**Pull request merged**